src/API.jl

```diff
@@ -570,26 +570,43 @@ function status(ctx::Context, mode=PKGMODE_PROJECT)
 end
 
 activate() = (Base.ACTIVE_PROJECT[] = Base.load_path_expand(nothing))
-function activate(path::String)
-    devpath = nothing
-    env = Base.active_project() === nothing ? nothing : EnvCache()
-    if env !== nothing && haskey(env.project["deps"], path)
-        uuid = UUID(env.project["deps"][path])
-        info = manifest_info(env, uuid)
-        devpath = haskey(info, "path") ? joinpath(dirname(env.project_file), info["path"]) : nothing
-    end
-    # `pkg> activate path`/`Pkg.activate(path)` does the following
-    # 1. if path exists, activate that
-    # 2. if path exists in deps, and the dep is deved, activate that path (`devpath` above)
-    # 3. activate the non-existing directory (e.g. as in `pkg> activate .` for initing a new env)
-    if Types.isdir_windows_workaround(path)
-        path = abspath(path)
-    elseif devpath !== nothing
-        path = abspath(devpath)
+function activate(path::String; shared::Bool=false)
+    if !shared
+        devpath = nothing
+        env = Base.active_project() === nothing ? nothing : EnvCache()
+        if env !== nothing && haskey(env.project["deps"], path)
+            uuid = UUID(env.project["deps"][path])
+            info = manifest_info(env, uuid)
+            devpath = haskey(info, "path") ? joinpath(dirname(env.project_file), info["path"]) : nothing
+        end
+        # `pkg> activate path`/`Pkg.activate(path)` does the following
+        # 1. if path exists, activate that
+        # 2. if path exists in deps, and the dep is deved, activate that path (`devpath` above)
+        # 3. activate the non-existing directory (e.g. as in `pkg> activate .` for initing a new env)
+        if Types.isdir_windows_workaround(path)
+            fullpath = abspath(path)
+        elseif devpath !== nothing
+            fullpath = abspath(devpath)
+        else
+            fullpath = abspath(path)
+        end
     else
-        path = abspath(path)
+        # this regex disallows names such as "Foo/bar", ".", "..", "Foo*", etc
+        occursin(r"([\\/:*?<>|])|(^\.+$)", path) && cmderror("not a valid name for a shared environment: $(path)")
```

Comment:
I'm against using regex here because regex is hard to read and it is difficult to determine all the edge cases across the different OS we support. I would prefer if we first determine depot and then:
```
basename(joinpath(depot,path)) == path # error out if this is false
```
I think it reflects the intention more directly.

Reply:
That wouldn't catch `.` or `..` though. but maybe that's not the responsibility of this API

Reply:
or I could do the following, which does catch `.` and `..`

```
basename(abspath(joinpath(fullpath,path))) == path
```

I'll try it